### PR TITLE
Add share_localhost.sh

### PR DESCRIPTION
**Pull request by medvedevthedev**
`share_localhost.sh` activates the virtual environment, runs the `share_localhost_qr.py` script, and then deactivates the virtual environment.

`share_localhost_qr.py` generates a QR code that contains the URL for the local network ip of the machine running the bot at port `8000` (which is where we run the default `BaseBot` server via `uvicorn`) and displays the QR code image.

Additionally, this PR cleans up `basebot_install.sh` and its dependencies.

---

**TL;DR** improve devx by outputting the local network ip of the `uvicorn` server to a qr code image and open it, clean up `basebot_install.sh`

### CODE REVIEW

File: scripts/tests.sh [hunk 1]
```diff
@@ -1,42 +1,45 @@
+#!/bin/bash
+
 # create tests directory and test case file
 mkdir tests
 touch tests/test_main.py
 
 # add continuity test
-echo '
+cat << EOF > tests/test_main.py
```
Comment (sergeybok) [inline scripts/tests.sh]: There's no reason why we should define this script in a string inside of a shell script. This will make modifying and testing this script very annoying. Much easier to have a python file.

File: scripts/tests.sh [hunk 2]
```diff
@@ -1,42 +1,45 @@
...
+parser.add_argument("--port", type=int, help="localhost:{port}", default=8000)
 args = parser.parse_args()
 
-bot_name = input('BotName: ')
-local_uuid = 'test1f3a-97ee-4e7e-8242-7b1d202c0fb5'
-url = f'http://localhost:{args.port}/bots/{bot_name}'
-resp = requests.get(url+'/about').json()
-bot_id = resp['bot_id']
+bot_name = input("Which bot are you trying to test?\n\nBotName: ")
+local_uuid = str(uuid.uuid4())
```
Comment (sergeybok) [inline scripts/tests.sh]: Having a static uuid makes the db not bloat, and you can easily clear all of the test messages. As well as use the same id in other test scripts.

File: scripts/tests.sh (hunk 1 above)
Comment (medvedevthedev) [reply scripts/tests.sh]: Pulling from `scripts/test.py` now, deleted `tests.sh`.

**Review by sergeybok — APPROVED** [review]
I tried to put the questions into the install and make the other changes that I'm saying now, but it caused more bugs and I don't want to deal with them now. Overall I feel that start_bot should only start a bot. and start_localhost should only find the ip and start the qr script. The install of qrcode lib should be in the main install. And the question of whether you want to start a bot should be in the install script. As well as the question of whether to start the test script should be in install and not in start_bots.

But it works so I'm approving. We can possibly change it later.

File: scripts/share_localhost.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import qrcode
+import subprocess
+
+# Get the hostname and IP (non loopback) address of the current machine
+cmd = "ifconfig | grep 'inet ' | grep -Fv 127.0.0.1 | awk '{print $2}'"
+output = subprocess.check_output(cmd, shell=True)
+ip_address = output.decode().strip()
+
+# Generate a QR code containing the URL for localhost
+url = f"http://{ip_address}:8000"
```
Comment (sergeybok) [inline scripts/share_localhost.py:10]: Would be better if it took as argument a name of the bot and did the direct address http://localhost:8000/bots/MyBot 

And no arg is http://localhost:8000/bots/

File: scripts/share_localhost.sh [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+# Define the name of the virtual environment
+venv_name="venv"
+
+# Check if the virtual environment exists
+if [ ! -d "$venv_name" ]; then
+    # If not, create it
+    echo "Creating virtual environment..."
+    python3 -m venv "$venv_name"
+fi
```
Comment (sergeybok) [inline scripts/share_localhost.sh]: This should be removed. This script should do one thing.

File: scripts/share_localhost.sh [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+# Activate the virtual environment
+echo "Activating virtual environment..."
+source "$venv_name/bin/activate"
+
+# Check if the required Python packages are installed
+if ! python3 -c "import qrcode" &> /dev/null; then
+    # If not, install them using pip
+    echo "Installing qrcode..."
+    pip install -q qrcode
+fi
```
Comment (sergeybok) [inline scripts/share_localhost.sh]: Also should be moved to where all other installs happen

File: scripts/start_bots.sh [hunk 6]
```diff
@@ -1,14 +1,42 @@
 #!/bin/bash
 
-# Start the app in the background and save the logs to a file
-nohup uvicorn main:app --port 8000 --host 0.0.0.0 > bot.log 2>&1 &
+read -p "Start demo bot [y/N]? " choice
```
Comment (sergeybok) [inline scripts/start_bots.sh]: I think this question should be in the install script. When you run start_bot it should start the bot.

File: scripts/start_bots.sh [hunk 7]
```diff
@@ -1,14 +1,42 @@
...
+    case "$choice" in 
+      y|Y )
+        # Encode local bot url into a qr code
+        echo "Encoding local bot url into a qr code..."
+        source venv/bin/activate
+        sh ./scripts/share_localhost.sh
+        deactivate
+        # Start test script
+        echo "Starting test script..."
+        python3 ./tests/test.py
```
Comment (sergeybok) [inline scripts/start_bots.sh]: This should also not be here. Maybe we can have a combined script that does start_and_test_bot.

**Review by sergeybok — APPROVED** [review]

File: scripts/share_localhost.py (hunk 3 above)
Comment (medvedevthedev) [reply scripts/share_localhost.py:10]: It's not `localhost` since we want to have other devices on the network (not apps on the same device that's running the bot server) connect to the bot by scanning the qr.

File: scripts/start_bots.sh (hunk 7 above)
Comment (medvedevthedev) [reply scripts/start_bots.sh]: Maybe we can add a check whether the user wants to generate a qr code or not? Not sure if there is a use case for just generating a qr code that encodes a given bot's local network url unless it's meant to be used since the device ip can change over time.

**Comment by medvedevthedev** [comment]
> I tried to put the questions into the install and make the other changes that I'm saying now, but it caused more bugs and I don't want to deal with them now. Overall I feel that start_bot should only start a bot. and start_localhost should only find the ip and start the qr script. The install of qrcode lib should be in the main install. And the question of whether you want to start a bot should be in the install script. As well as the question of whether to start the test script should be in install and not in start_bots.
> 
> But it works so I'm approving. We can possibly change it later.

Made the requested changes and verified that they work – only thing I did not change was `start_bots.sh` asking to start the test script. We can make a dedicated script for that, or just add instructions for calling `test.py` to our `README` in a separate PR.
